test/basic-querying.test.js

```diff
@@ -127,7 +127,23 @@ describe('basic-querying', function() {
 
   describe('find', function() {
 
-    before(seed);
+    before(function() {
+      seed(function setupDelayingLoadedHook() {
+        User.observe('loaded', function(ctx, next) {
+          var randomTimeoutTrigger = Math.floor(Math.random() * 100);
+
+          setTimeout(function() {
+            process.nextTick(next);
+          }, randomTimeoutTrigger);
+        });
+      });
+    });
+
+    after(function removeDelayingLoadHook() {
+      User.observe('loaded', function(ctx, next) {
```

Comment:
It is possible to register multiple hook observers, this code does not remove the hook registered in `before` hook. You should call `removeObserver` API instead. It looks we don't include the observer API in the apidocs at apidocs.strongloop.com, here is the source code: https://github.com/strongloop/loopback-datasource-juggler/blob/2189b9a746071b599d6a129313b2cdd11052e36b/lib/observer.js#L45

I think the usage be like this:

```js
// put this function somewhere at the bottom of this file, where other helper methods are
function nextAfterDelay(ctx, next) {
  var randomTimeoutTrigger = Math.floor(Math.random() * 100);
  setTimeout(function() { process.nextTick(next); }, randomTimeoutTrigger);
}

before(seed);
before(function setupDelayingLoadedHook() {
  User.observe('loaded', nextAfterDelay);
});

after(function removeDelayingLoadedHook) {
  User.removeObserver('loaded', nextAfterDelay);
});
```